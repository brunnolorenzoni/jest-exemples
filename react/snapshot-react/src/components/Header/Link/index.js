import React, { useState } from 'react';


const Link = (props) => {

    const { href } = props;

    const [ classHover, setClassHover ] = useState('')

    const handleMouseEnter = () => {
        setClassHover('hover');
    }

    const handleMouseLeave = () => {
        setClassHover('');
    }


    return (
        <a href={href} 
            className={"link " + classHover} 
            onMouseEnter={handleMouseEnter}
            onMouseLeave={handleMouseLeave}
        >
            Home
        </a>
    )
}

export default Link;