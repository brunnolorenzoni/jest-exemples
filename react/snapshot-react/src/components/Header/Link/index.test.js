import React from 'react';
import Link from './index';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';

test('Link changes the class when hovered', () => {
    const component = renderer.create(
      <Link href="/contacts">Contacts</Link>,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
    // manually trigger the callbawck
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    
  
    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });