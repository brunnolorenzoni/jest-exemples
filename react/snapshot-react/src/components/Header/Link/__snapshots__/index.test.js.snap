// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<a
  className="link "
  href="/contacts"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`changes the class when hovered 2`] = `
<a
  className="link hover"
  href="/contacts"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`changes the class when hovered 3`] = `
<a
  className="link "
  href="/contacts"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`renders as an anchor when no page is set 1`] = `
<a
  className="link "
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`renders correctly 1`] = `
<a
  className="link "
  href="/contacts"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;
